vector/src/main/java/im/vector/app/features/signout/soft/SoftLogoutViewModel.kt

```diff
@@ -56,15 +57,23 @@ class SoftLogoutViewModel @AssistedInject constructor(
     companion object : MavericksViewModelFactory<SoftLogoutViewModel, SoftLogoutViewState> by hiltMavericksViewModelFactory() {
 
         override fun initialState(viewModelContext: ViewModelContext): SoftLogoutViewState? {
-            val activity: SoftLogoutActivity = (viewModelContext as ActivityViewModelContext).activity()
-            val userId = activity.session.myUserId
-            return SoftLogoutViewState(
-                    homeServerUrl = activity.session.sessionParams.homeServerUrl,
-                    userId = userId,
-                    deviceId = activity.session.sessionParams.deviceId ?: "",
-                    userDisplayName = activity.session.getUser(userId)?.displayName ?: userId,
-                    hasUnsavedKeys = activity.session.hasUnsavedKeys()
-            )
+            val sessionHolder = EntryPoints.get(viewModelContext.app(), SingletonEntryPoint::class.java)
```

Comment:
I'm not sure this is the correct way to inject into the initial state, we can provide other instances via https://github.com/vector-im/element-android/pull/5258/files#diff-a57f93c54f89c6899826efa9c728f09e6172d330131c1ae6a33e2ac1cfb26a5fR54

Reply:
I copy/paste the same logic I found in other `initialState(viewModelContext: ViewModelContext)` methods. But I don't understand your suggestion. Can you explain it more, please?

Reply:
ah no worries, I didn't realise `hiltMavericksViewModelFactory` was our own helper which also directly interacts with the `EntryPoints`

I was imagining the `@Assisted initialState: SoftLogoutViewState` would allow us to do... 

```kotlin
class Factory(activeSessionHolder) {
  fun create() = InitialState(activeSessionHolder.foo)
}
```